src/rules/typing/intent.rs

```diff
@@ -0,0 +1,153 @@
+        // Names of all the dummy arguments
+        let parameters: Vec<&str> = node
+            .child_by_field_name("parameters")?
+            .named_children(&mut node.walk())
+            .filter_map(|param| Some(to_text(&param, &src)?))
+            .collect();
+
+        let parent = node.parent()?;
+        let procedure_kind = parent.kind();
+
+        // Logic here is:
+        // 1. find variable declarations
+        // 2. filter to the declarations that don't have an `intent`
+        // 3. filter to the ones that contain any of the dummy arguments
+        // 4. collect into a vec of violations
+        //
+        // We filter by missing intent first, so we only have to
+        // filter by the dummy args once -- otherwise we either catch
+        // local var decls on the same line, or need to iterate over
+        // the decl names twice
+        let violations = parent
+            .named_children(&mut parent.walk())
+            .filter(|child| child.kind() == "variable_declaration")
+            .filter(|decl| {
+                !decl
+                    .children_by_field_name("attribute", &mut decl.walk())
+                    .any(|attr| {
+                        to_text(&attr, &src)
+                            .unwrap_or("")
+                            .to_lowercase()
+                            .starts_with("intent")
+                    })
+            })
+            .filter_map(|decl| {
+                let violations: Vec<Violation> = decl
+                    .children_by_field_name("declarator", &mut decl.walk())
+                    .filter_map(|declarator| {
+                        let identifier = match declarator.kind() {
+                            "identifier" => Some(declarator),
+                            "sized_declarator" => child_with_name(&declarator, "identifier"),
+                            // Although tree-sitter-fortran grammar allows
+                            // `init_declarator` and `pointer_init_declarator`
+                            // here, dummy arguments aren't actually allow
+                            // initialisers. _Could_ still catch them here, and
+                            // flag as syntax error elsewhere?
+                            _ => None,
+                        }?;
+                        let name = to_text(&identifier, &src)?;
+                        if parameters.contains(&name) {
+                            return Some((declarator, name));
+                        }
+                        None
+                    })
+                    .filter_map(|(dummy, name)| {
+                        let msg = format!(
+                            "{procedure_kind} argument '{name}' missing 'intent' attribute"
+                        );
+                        Some(Violation::from_node(&msg, &dummy))
+                    })
+                    .collect();
+                Some(violations)
```

Comment:
```suggestion
                violations
```

... and the next one...